### PR TITLE
milestone 1: Creating data model

### PR DESCRIPTION
**Pull request by bronnie313**
## Blog app - creating a data model

- Created a new app called the blog app and initialized it with git
- Configured it to use the Postgres database 
- Built schema using the migration files 
- Added indexes to corresponding foreign keys in the columns

### CODE REVIEW

File: db/migrate/20230816154551_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      t.integer :author_id
+      t.string :title
+      t.text :text
+
+      t.timestamps
+      t.integer :comments_counter, default: 0
+      t.integer :likes_counter, default: 0
+    end
+
+    add_foreign_key :posts, :users, column: :author_id
```
Comment (rloterh) [inline db/migrate/20230816154551_create_posts.rb:13]: - [ ] `Author` is aliasing `User` model and thus we have `AuthorId` used as reference for associated models to `User`, as seen in the given [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Hence, our migration query should be similar to:

```ruby
...
  def change
    add_reference :posts, :author, references: :users, null: false, foreign_key: {to_table: :users}
  end
...

```

or.. 

```ruby
  create_table :posts do |t|
    . . .
    t.references :author, foreign_key: { to_table: 'users' } 
    . . .
  end
```

File: db/migrate/20230816155937_create_comment.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  def change
+    create_table :comments do |t|
+      t.integer :user_id
+      t.integer :post_id
+      t.text :text
+
+      t.timestamps
+    end
+
+    add_foreign_key :comments, :users, columm: :user_id
```
Comment (rloterh) [inline db/migrate/20230816155937_create_comment.rb:11]: - [ ] Similarly, `Author` is aliasing `User` model and thus we have `AuthorId` used as reference for associated models to `User`, as seen in the given [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Hence, our migration query should be similar to:

```ruby
...
  def change
    add_reference :comments, :author, references: :users, null: false, foreign_key: {to_table: :users}
  end
...

```

or.. 

```ruby
  create_table :comments do |t|
    . . .
    t.references :author, foreign_key: { to_table: 'users' } 
    . . .
  end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Bisaso, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 
🚦 __🏃 🏡 🏃‍♂️ 🏚️ 🌴 🗼

## Highlights 👍
- Good job so far creating migrating query!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — APPROVED** [review]
Hi @bronnie313 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

chelsea-mason-approve


## Highlights
Great effort in making the requested changes :+1: 
Neat and well organized code :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**